components/requests/ProcessingTasksCard.tsx

```diff
@@ -0,0 +1,143 @@
+import { Divider, VStack, Button, Text, Link } from '@chakra-ui/react'; // Chakra UI
+import PermitHolderInfoCard from '@components/internal/PermitHolderInfoCard'; // Custom Card Component
+import AssignNumberModal from '@components/requests/modals/AssignNumberModal'; // AssignNumber Modal component
+import ProcessingTaskStep from '@components/requests/ProcessingTaskStep'; // Processing Task Step
+import { useState } from 'react'; // React
+
+type ProcessingTasksCardProps = {
+  readonly applicationProcessingStepsCompleted: number[];
+  readonly onTaskComplete: (taskId: number) => void;
+  readonly onTaskUndo: (taskId: number) => void;
+};
+
+export default function ProcessingTasksCard({
+  applicationProcessingStepsCompleted,
+  onTaskComplete,
+  onTaskUndo,
+}: ProcessingTasksCardProps) {
+  const [APPNumber, setAPPNumber] = useState<number | undefined>(undefined);
+  const [invoiceNumber, setInvoiceNumber] = useState<number | undefined>(undefined);
+
+  const assignAPPNumber = (APPNumber: number) => {
+    onTaskComplete(1);
+    setAPPNumber(APPNumber);
+  };
+
+  const assignInvoiceNumber = (invoiceNumber: number) => {
+    onTaskComplete(4);
+    setInvoiceNumber(invoiceNumber);
+  };
+
+  const steps = [
+    // Task 1: Assign new APP number: Assign number (MODAL)
+    {
+      label: 'Assign new APP number' + (APPNumber ? `: ${APPNumber}` : ''),
+    },
+    // Task 2: Hole punch parking permit: Mark as complete (CHECK)
+    {
+      label: 'Hole punch parking permit',
+      description: 'Gender, Year and Month',
+    },
+    // Task 3: Create a new wallet card: Mark as complete (CHECK)
+    {
+      label: 'Create a new wallet card',
+      description: 'Include permit number, expiry date, full name, and birth month',
+    },
+    // Task 4: Assign invoice number: Assign number (MODAL)
+    {
+      label: 'Assign invoice number' + (invoiceNumber ? `: ${invoiceNumber}` : ''),
+    },
+    // Task 5: Upload document: Choose document (UPLOAD FILE)
+    {
+      label: 'Upload document',
+      description: 'Scan all documents and upload as PDF (5MB limit)',
+    },
+    // Task 6: Mail out: Mark as complete (CHECK)
+    {
+      label: 'Mail out',
+      description: 'Include returning envelope and previous permit number',
+    },
+  ];
+
+  const taskAction = (taskId: number, isCompleted: boolean) => {
```

Comment:
```suggestion
  const _renderTaskAction = (taskId: number, isCompleted: boolean) => {
```
This f unction is used for rendering, so let's make that clear. Might also be helpful to add a JSDoc comment